### ISSUE TITLE
Users should be able to 'like' posts.

### ISSUE THREAD

**Issue by scd** [theCrag/website#3130]
**What happened?**

See this discussion where people have requested that they can like posts.

https://www.thecrag.com/event/1729796049

**Comment by scd** [comment]
Any solution should consider liking other things such as photos.

It should also consider more generically 'reacting' to a post.

**Comment by brendanheywood** [comment]
Yes totally agree under the hood this should be exactly the same as liking photos:

https://github.com/theCrag/website/issues/1293

Some quick thoughts:

1) I really like the idea of the reactions being a very limited set of climbing specific, eg 'fun' :laughing:  'strong' :muscle: , 'respect' :fist_oncoming:    , 'scary' :fearful:  and trying to avoid directly negative ones

2) What is the scope of a reaction? Do you react to the stream event, or the ascent, any stream item like an edit. Or do we also allow it on any comment as well? I'm leaning towards all of them

3) What is the function of a reaction? I think for most contexts a reaction is functionally the same as commenting in that you get subscribed to the thread.

4) Privacy settings, there is functionally not really any difference between somebody reacting to an ascent with :fearful: vs them commenting and pasting that same emoji into the comment. But perception wise there may be, so this may issue be the catalyst for adding more fine grained permissions, eg saying not only who can see my ascents but who can comment / react to them

**Comment by goshenw** [comment]
Thought I'd sign up to add a comment on this one.   As noted in the above thread, there is some concern over the ability to "Like" Ascents.    There may be no technical difference between liking, and placing a comment with an emoticon; but there is an 'effort' difference.

I think the fear is that liking ascents will just divulge into a social shitfight, where users are become concerned or about how many likes their ascents are getting (or not getting).     EG: If someone makes the effort to comment on one of my ascents (quite rare); then I am stoked.   If you introduce the ability to "Like" ascents, and I see that 'other' people are getting likes, but 'mine' are not, it would be depressing.  We've got Facebook for that kind of social media anxiety!!

Liking comments is, I think different, in that you are lending your support / opinion to a conversation  without adding needless "I agree" comments to the forum.

Goshen

**Comment by brendanheywood** [comment]
@goshenw lets tease this apart a little more:

I still don't see much difference between a comment saying :muscle: and a 1-click reaction button saying :muscle:. If adding it as a convenient 1-click button makes it twice as likely that people will do it, then in theory the ratio of reactions to non-reactions isn't really going to change.

So are you concerned that my 'ratio' assumption above is actually wrong, ie by making it easier then someone who doesn't get many reactions might get double their normal reactions, but someone who was previously getting more manual reactions might now get exponentially more reactions and it snowballs and creates a massive difference?

I'm also curious about the mental perception of "I don't have any likes" vs "I don't have any comments". I could see how the lack of comments is neutral, but the lack of likes is kinda equated to "Nobody likes me". That's why I've been trying to steer away from even having 'like' and just having things like 'strong!' and 'scary!'.

So perhaps we could have only reactions (strong, fun, scared, wow) at the stream and ascents / photo level, and then for comments we'd only have something simpler  eg 'like'. It probably doesn't make much sense to 'strong' a comment. I'm a little meh about adding the idea of an 'angry' or 'sad' reaction at either level, but we probably need some way for people to express disagreement with a comment if they are also able to agree with a comment. So maybe at the comment level it's just a thumbs up / thumbs down.

**Comment by scd** [comment]
Would I be correct in assuming that the number of likes in a post in facebook is tied to somebodies ego. If you don't have very many likes it hurts your ego.

theCrag has been a fairly social safe zone - unless you use the wrong tick type :). 

Is it possible that the rareness of comments on an ascent keeps theCrag in this social safe zone where you do not judge yourself on how many comments you get for a tick. But importantly there may be a critical mass of comments where you now judge yourself on the number of comments you get.

If this is the case then the ratio does not matter nor does the like button. The like button simply propels theCrag to the critical mass of comments where ego comes in.

**Comment by AndreasAachen** [comment]
I agree 100% with scd's comment above. 

> Is it possible that the rareness of comments on an ascent keeps theCrag in this social safe zone where you do not judge yourself on how many comments you get for a tick. 
<-- Absolutely my opnion. People rarely comment, and a lot of the time the comments are rather usefull or have some questions or discussion in them (in contrast to FB where basically anything usefull dissapears in the noise). 

I think theCarg has managed to allow people to comment/share their opinions on ticks of others without drifting too far into the social media direction (as Facebook or Instagramm). The addition of a like button (or whatever you want to call it), will change this. 

Maybe instead of a likebutton on TheCrag just add a 'share this tick to FB button'. If people are really keen for likes etc. they can get them there.

Moreover, I also agree with Goshen, that a like button for comments could be very usefull. As it is a quick way to communicate and highlight good ideas and thoughts in the comments section. (Even the new york times does something similar in the comment section of articles).

**Comment by goshenw** [comment]
> The like button simply propels theCrag to the critical mass of comments where ego comes in.
Yes, I think this might be the case...  I also see how it might increase repeat traffic to the crag :-)  (Not saying that's your motivation).
>So maybe at the comment level it's just a thumbs up / thumbs down.
Great idea - I've seen that work well elsewhere.

BTW   I'm not so concerned about my own social feed (likes vs whatever; I'm over that); but just the general 'feel' of social media and how it plays with our physiology, but i'm no expert.

However on facebook, when someone posts, they make one post, and you like it or whatever, but when you log ascents; you're logging 5 or 20; that's 20 opportunities to like or fistbump (I do like the fistbump idea - leave it at that if you do anything).   I just question whether it ads a layer that doesn't need to be there.   Obviously people aren't going to fist bump each ascent.   But as a user, am I going to be put in a position of having to make that 'choice' about each of my friends ascents, and whether they deserve a fist bump or not.   Going back to it; a universal ability to 'comment' is not the same thing, IMO.

But I won't harp on about it; it's just the way I see things, and perhaps can be considered in your final decision.

**Comment by scd** [comment]
My take on this that we should at least be careful what we do with ticks as the logbook is sacred. Like away on everything else, just don't touch my logbook ticks. This has come up before in other enhancements. 

We will do some wider consultation before we implement. I think we want a solution which enhances connectivity of the community but not the social ego. It will be a while before we implement anything here as we have a tonne of priorities.

**Comment by Drazhar** [comment]
I understand the concerns of liking ascents and comments, but liking photos would help a lot.

First it would give a feeling which photos should be shown more often at an area but moreover it would encourage people to upload more images of an area, which is in general a good thing. The only reason Instagram and Co gets so much images is the ego of the people, TheCrag should use this also IMHO.

**Comment by Drazhar** [comment]
Also the possibility to like if someone is editing in a good way, would also encourage the people to edit more often.

Even if commenting is just 2 clicks more then liking, the inhibitions to do it is much higher in general.

**Comment by brendanheywood** [comment]
Another +1 

https://www.thecrag.com/discussion/1812000417/aussie-version-of-alex-honnold#m1813643622

**Comment by DaneEvans** [comment]
Not sure that giving the ability to like based on that particular thread is a good idea... 

speaking of which, is @commonsensebolter just a troll, private account, and from what I can see, no previous history of being a user of the site.

**Comment by brendanheywood** [comment]
+1 from support email

**Comment by Luen** [comment]
+1 for like button on images, etc

**Comment by bonnie-zhang** [comment]
+1 for allowing likes/reactions on comments

**Comment by brendanheywood** [comment]
+1 from https://www.thecrag.com/discussion/2490574413/inconsistentcies#m2493162552
